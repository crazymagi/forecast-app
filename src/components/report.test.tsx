import { render, screen, waitFor, within } from '@testing-library/react';
import axios from "axios";
import Report from './report';
import mockData from "../mock-data/report-mockdata.json";
import { act } from 'react-dom/test-utils';

jest.mock('axios');

beforeEach(() => {
  (axios as any).get.mockResolvedValue({ status: 200, data: mockData });
})

test('renders matches snapshot', () => {
  const {asFragment} = render(<Report latitude={44.44} longitude={55.55} days={7} />)
  expect(asFragment()).toMatchSnapshot();
});

test.each(['Date', 'Max Temp', 'Min Temp'])('renders data headers', (header) => {
  render(<Report latitude={44.44} longitude={55.55} days={7} />);
  const headerElement = screen.getByText(header);
  expect(headerElement).toBeInTheDocument();
});

test('renders expected rows', async () => {
  render(<Report latitude={44.44} longitude={55.55} days={7} />);
  
  expect(screen.getByText("Loading...")).toBeInTheDocument();

  await waitFor(() => {
    expect(
      screen.queryByText('Loading...'),
    ).not.toBeInTheDocument()
  })

  const rows = await screen.findAllByRole('row');
  expect(rows).toHaveLength(8);

  const dataRow = rows[1];
  const dateElement = within(dataRow).getByText('2023-07-10')
  expect(dateElement).toBeInTheDocument();
  const maxTempElement = within(dataRow).getByText('27.7')
  expect(maxTempElement).toBeInTheDocument();
  const minTempElement = within(dataRow).getByText('19.1')
  expect(minTempElement).toBeInTheDocument();
});
