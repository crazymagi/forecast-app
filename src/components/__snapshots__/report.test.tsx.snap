// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg MuiTableContainer-root css-3o31d6-MuiContainer-root-MuiTableContainer-root"
  >
    <div>
      Loading...
    </div>
    <table
      aria-label="forecast table"
      class="MuiTable-root css-ksyw7d-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ndpvdd-MuiTableCell-root"
            scope="col"
          >
            Date
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ndpvdd-MuiTableCell-root"
            scope="col"
          >
            Max Temp
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ndpvdd-MuiTableCell-root"
            scope="col"
          >
            Min Temp
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      />
    </table>
  </div>
</DocumentFragment>
`;
